lib/vrl/stdlib/src/parse_url.rs

```diff
@@ -44,32 +69,47 @@ impl Function for ParseUrl {
         mut arguments: ArgumentList,
     ) -> Compiled {
         let value = arguments.required("value");
+        let default_known_ports = arguments.optional("default_known_ports");
```

Comment:
nit: I usually apply the pattern of `.unwrap_or_else(|| expr!(false))`. Which allows you to have `ParseUrlFn` to store a regular `Box<dyn Expression>`, avoiding the more verbose option-dance in the expression trait impl.

Reply:
Nice, I wasn't aware of this pattern. It's definitely more concise.